tmt/utils.py

```diff
@@ -964,6 +964,10 @@ def public_git_url(url):
     authentication. For now just cover the most common services.
     """
 
+    # Using internal url instance transformtion is not supported
+    if re.match(r'ssh://.*osci\.redhat\.com/.*', url):
+        raise ConvertError(f"Use distgit, transformation for {url} is not supported")
+
     # GitHub, GitLab
```

Comment:
if this is about gitlab/github, we should rather match below gitlab/github, it is not tied to gitlab/github at all in the regex ....

Reply:
At all this trasnformation probably should be just for page where we know it is possible to trasform as you said. and aovid transform any url from some protocol to `https`

But still not sure if there is good way how to do it. maybe when doing this transformation to check if the target is accessible.
If not raise error or use non-transformed url. But this check of external resources will cost some time.

Reply:
We have #481 for external resources - since this happens sporadically it is better to play safe and make sure reference is correct

Reply:
And with that said - maybe is this check obsolete? Because we need to assert whole fmf-id

Reply:
@lukaszachy Theoretically it is obsolote in case it also checks who fmf-id, but in fact. it is closer to git and transformations, which are not sometimes correct and it is better to check.

But finally my solution helps to solve chicken-egg problem. because it is not good to commit changes to git, without calling tmt test export --nitrate --general before, because it modifies fmf files, and also it is not possible to export it firstly, because in case you'll try to check whole `fmf_id` proper data could not be commited to git, so that
and catch it early

At least for export it could be problem to check full fmf_id identifier.

For lint it is okay.

Reply:
I beg to differ: It is easy to run tmt export but forgot to run git push and that is thing we have to catch.

Reply:
@lukaszachy right.This PR is very simple and just close to transformation and find it early that final fmf_id url will be bad.
But in fact check full fmf identifier is superset of this simple check.

Yes, this commit&push miss is bad, but you cannot do that direclty for export, because it is expected to not have it commited and pushed to git in export phase, but just via some tmt test lint or what, but you can also forgot to call `fmf lint` so that it does not solve this issue.

For your issue is better to print some message at the end with **big red warning** do not forget to to commit and push changes to git, after export.

Reply:
> ..., because it is expected to not have it commited ...

Not sure. While a bit annoying (two commits) we shouldn't lead user to create race in TCMS.

Reply:
@lukaszachy exactly, so that for export you are not able to check full fmf_id check as it does not exists inside remote URL without commit and push, so this PR helps you to check at least `url` what is probably the worst case.